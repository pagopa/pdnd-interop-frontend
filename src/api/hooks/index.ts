export * from './useDownloadFile'
export * from './useAuthenticatedQuery'
