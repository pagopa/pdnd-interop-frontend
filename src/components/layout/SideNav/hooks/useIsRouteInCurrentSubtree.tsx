import { type RouteKey, getParentRoutes, useCurrentRoute } from '@/router'

export function useIsRouteInCurrentSubtree() {
  const { routeKey: currentRouteKey } = useCurrentRoute()

  console.log('currentRouteKey', currentRouteKey)
  return (routeKey: RouteKey) => {
    return [...getParentRoutes(currentRouteKey), currentRouteKey].includes(routeKey)
  }
}
