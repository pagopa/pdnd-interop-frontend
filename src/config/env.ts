import type { ExtendedWindow, PagoPAEnvVars } from '@/types/common.types'

const PAGOPA_ENV = (window as unknown as ExtendedWindow).pagopa_env

export const isDevelopment = !!(import.meta.env.MODE === 'development')
export const isProduction = !!(import.meta.env.MODE === 'production')
export const isTest = !!(import.meta.env.MODE === 'test')

if (!PAGOPA_ENV && !isTest) {
  console.warn('pagopa_env not available.')
}

const DEV_API_HOST_URL = import.meta.env.REACT_APP_API_HOST
const DEV_SELFCARE_LOGIN_URL = import.meta.env.REACT_APP_SELFCARE_LOGIN_URL
const DEV_INTEROP_RESOURCES_BASE_URL = import.meta.env.REACT_APP_INTEROP_RESOURCES_BASE_URL
const DEV_ONETRUST_DOMAIN_SCRIPT_ID = import.meta.env.REACT_APP_ONETRUST_DOMAIN_SCRIPT_ID

export const FE_LOGIN_URL = (
  isProduction && PAGOPA_ENV ? PAGOPA_ENV.SELFCARE_LOGIN_URL : DEV_SELFCARE_LOGIN_URL
) as string
export const PUBLIC_URL = import.meta.env.PUBLIC_URL || '/ui'
export const FE_URL = `${window.location.origin}${PUBLIC_URL}`

export const INTEROP_RESOURCES_BASE_URL =
  isProduction && PAGOPA_ENV
    ? PAGOPA_ENV.INTEROP_RESOURCES_BASE_URL
    : DEV_INTEROP_RESOURCES_BASE_URL
export const ONETRUST_DOMAIN_SCRIPT_ID = (
  isProduction && PAGOPA_ENV ? PAGOPA_ENV.ONETRUST_DOMAIN_SCRIPT_ID : DEV_ONETRUST_DOMAIN_SCRIPT_ID
) as string
export const MIXPANEL_PROJECT_ID = isProduction && PAGOPA_ENV ? PAGOPA_ENV.MIXPANEL_PROJECT_ID : ''

export const TEMP_USER_WHITELIST_URL = `${INTEROP_RESOURCES_BASE_URL}/temp-whitelist.json`

function getEnvVar(varName: keyof PagoPAEnvVars, devVarName: string): string {
  return isProduction && PAGOPA_ENV && PAGOPA_ENV
    ? PAGOPA_ENV[varName]
    : `${DEV_API_HOST_URL}/${devVarName}`
}

const SERVICE_VERSION = import.meta.env.REACT_APP_SERVICE_VERSION
export const API_GATEWAY_INTEFACE_URL = getEnvVar(
  'API_GATEWAY_INTEFACE_URL',
  'swagger/docs/interface-specification.yml'
)
export const BACKEND_FOR_FRONTEND_URL = getEnvVar(
  'BACKEND_FOR_FRONTEND_URL',
  `backend-for-frontend/${SERVICE_VERSION}`
)
export const AUTHORIZATION_PROCESS_URL = getEnvVar(
  'AUTHORIZATION_PROCESS_URL',
  `authorization-process/${SERVICE_VERSION}`
)
export const AUTHORIZATION_SERVER_ACCESS_TOKEN_URL = getEnvVar(
  'AUTHORIZATION_SERVER_TOKEN_CREATION_URL',
  'authorization-server/token.oauth2'
)
export const CLIENT_ASSERTION_JWT_AUDIENCE =
  isProduction && PAGOPA_ENV ? PAGOPA_ENV.CLIENT_ASSERTION_JWT_AUDIENCE : ''

function getWellKnownUrls(wellKnownUrls: string | undefined) {
  return wellKnownUrls?.split(',').filter((url) => !!url) || []
}

export const WELL_KNOWN_URLS =
  isProduction && PAGOPA_ENV ? getWellKnownUrls(PAGOPA_ENV.WELL_KNOWN_URLS) : ['#']

export const SELFCARE_BASE_URL =
  isProduction && PAGOPA_ENV ? PAGOPA_ENV.SELFCARE_BASE_URL : 'https://uat.selfcare.pagopa.it'

export const STAGE = PAGOPA_ENV?.STAGE ?? 'DEV'

export const SELFCARE_INTEROP_PROD_ID = `prod-interop${STAGE === 'TEST' ? '-coll' : ''}`
