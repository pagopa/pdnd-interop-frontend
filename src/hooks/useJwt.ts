import React from 'react'
import { useAuth } from '@/stores'
import type { JwtUser } from '@/types/party.types'
import memoize from 'lodash/memoize'

const parseJwt = memoize((token: string | null) => {
  const jwt = token ? (JSON.parse(atob(token.split('.')[1])) as JwtUser) : undefined
  const currentRoles = jwt ? jwt.organization.roles.map((r) => r.role) : []
  const isAdmin = currentRoles.length === 1 && currentRoles[0] === 'admin'
  const isOperatorAPI = Boolean(currentRoles.includes('api'))
  const isOperatorSecurity = Boolean(currentRoles.includes('security'))

  return { jwt, currentRoles, isAdmin, isOperatorAPI, isOperatorSecurity }
})

export const useJwt = () => {
  const { sessionToken } = useAuth()

  const { jwt, currentRoles, isAdmin, isOperatorAPI, isOperatorSecurity } = parseJwt(sessionToken)

  const hasSessionExpired = React.useCallback(
    () => (jwt ? new Date() > new Date(jwt.exp * 1000) : false),
    [jwt]
  )

  const isCurrentUser = React.useCallback(
    (userId: string) => {
      return jwt && jwt.uid === userId
    },
    [jwt]
  )

  return {
    jwt,
    isCurrentUser,
    hasSessionExpired,
    isAdmin,
    isOperatorAPI,
    isOperatorSecurity,
    currentRoles,
  }
}
