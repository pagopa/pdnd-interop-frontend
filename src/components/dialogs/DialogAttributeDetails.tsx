import React from 'react'
import { Button, Dialog, DialogActions, DialogContent, DialogTitle, Stack } from '@mui/material'
import { useTranslation } from 'react-i18next'
import type { DialogAttributeDetailsProps } from '@/types/dialog.types'
import { useDialog } from '@/stores'
import { AttributeQueries } from '@/api/attribute'
import { InformationContainer, InformationContainerSkeleton } from '@pagopa/interop-fe-commons'

export const DialogAttributeDetails: React.FC<DialogAttributeDetailsProps> = ({ attribute }) => {
  const { t } = useTranslation('common')

  const ariaLabelId = React.useId()
  const { closeDialog } = useDialog()

  return (
    <Dialog aria-labelledby={ariaLabelId} open onClose={closeDialog} fullWidth>
      <DialogTitle id={ariaLabelId}>{attribute.name}</DialogTitle>

      <React.Suspense fallback={<AttributeDetailsSkeleton />}>
        <AttributeDetails attributeId={attribute.id} />
      </React.Suspense>

      <DialogActions>
        <Button variant="contained" onClick={closeDialog}>
          {t('closeBtn')}
        </Button>
      </DialogActions>
    </Dialog>
  )
}

const AttributeDetails: React.FC<{ attributeId: string }> = ({ attributeId }) => {
  const { t } = useTranslation('shared-components', {
    keyPrefix: 'dialogAttributeDetails',
  })
  const { data: attribute } = AttributeQueries.useGetSingle(attributeId)

  if (!attribute) return <AttributeDetailsSkeleton />

  return (
    <DialogContent>
      <Stack sx={{ mt: 1 }} spacing={2}>
        <InformationContainer
          content={attribute.description}
          direction="column"
          label={t('content.descriptionField.label')}
        />
      </Stack>
    </DialogContent>
  )
}

const AttributeDetailsSkeleton: React.FC = () => {
  return (
    <DialogContent>
      <Stack sx={{ mt: 1 }} spacing={2}>
        <InformationContainerSkeleton />
      </Stack>
    </DialogContent>
  )
}
