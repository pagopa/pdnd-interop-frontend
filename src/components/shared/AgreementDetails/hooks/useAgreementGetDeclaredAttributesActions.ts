import { isAttributeOwned } from '@/utils/attribute.utils'
import { useAgreementDetailsContext } from '../AgreementDetailsContext'
import { useTranslation } from 'react-i18next'
import { useCurrentRoute } from '@/router'
import { AttributeMutations } from '@/api/attribute'
import { AuthHooks } from '@/api/auth'

/**
 * Returns the actions for the declared attributes section inside the agreement details.
 * @returns A function that returns an array of actions for the given attribute.
 */
export const useAgreementGetDeclaredAttributesActions = () => {
  const { t } = useTranslation('agreement', { keyPrefix: 'read.attributes' })
  const { routeKey } = useCurrentRoute()
  const { isAdmin } = AuthHooks.useJwt()
  const { partyAttributes, isAgreementEServiceMine, agreement } = useAgreementDetailsContext()

  const { mutate: declareAttribute } = AttributeMutations.useDeclarePartyAttribute()

  return (attributeId: string) => {
    // The user can declare his own attributes only in the agreement create/edit view...
    if (!agreement || routeKey !== 'SUBSCRIBE_AGREEMENT_EDIT' || !isAdmin) return []
    if (isAgreementEServiceMine) return []
    const isDeclared = isAttributeOwned('declared', attributeId, partyAttributes?.declared ?? [])
    // ... only if it is not alread declared
    if (isDeclared) return []
    // ... and only if the agreement is active, draft or suspended
    if (!['ACTIVE', 'DRAFT', 'SUSPENDED'].includes(agreement.state)) return []

    const handleDeclareAttribute = (attributeId: string) => {
      declareAttribute({
        id: attributeId,
      })
    }

    return [
      {
        label: t('declared.actions.declare'),
        action: handleDeclareAttribute,
      },
    ]
  }
}
