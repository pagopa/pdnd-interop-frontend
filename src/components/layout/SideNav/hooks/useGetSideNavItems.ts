import React from 'react'
import { useJwt } from '@/hooks/useJwt'
import type { SideNavItemView } from '../SideNav'
import type { UserProductRole } from '@/types/party.types'
import EmailIcon from '@mui/icons-material/Email'
import type { RouteKey } from '@/router/router.types'
import uniq from 'lodash/uniq'

type Views = Record<UserProductRole, Array<SideNavItemView>>

const views: Views = {
  admin: [
    {
      routeKey: 'PROVIDE',
      id: 'provider',
      children: ['PROVIDE_ESERVICE_LIST', 'PROVIDE_AGREEMENT_LIST', 'PROVIDE_PURPOSE_LIST'],
    },
    {
      routeKey: 'SUBSCRIBE',
      id: 'subscriber',
      children: [
        'SUBSCRIBE_CATALOG_LIST',
        'SUBSCRIBE_AGREEMENT_LIST',
        'SUBSCRIBE_PURPOSE_LIST',
        'SUBSCRIBE_CLIENT_LIST',
        'SUBSCRIBE_INTEROP_M2M',
      ],
    },
  ],
  api: [
    {
      routeKey: 'PROVIDE',
      id: 'provider',
      children: ['PROVIDE_ESERVICE_LIST'],
    },
    {
      routeKey: 'SUBSCRIBE',
      id: 'subscriber',
      children: ['SUBSCRIBE_CATALOG_LIST'],
    },
  ],
  security: [
    {
      routeKey: 'SUBSCRIBE',
      id: 'subscriber',
      children: ['SUBSCRIBE_CATALOG_LIST', 'SUBSCRIBE_CLIENT_LIST', 'SUBSCRIBE_INTEROP_M2M'],
    },
  ],
}

export function useGetSideNavItems() {
  const { currentRoles } = useJwt()

  return React.useMemo(() => {
    const availableSideNavItems: Array<SideNavItemView> = []

    // For each user roles
    currentRoles.forEach((userRole) => {
      // ... take all the sidenav items for that role...
      const roleSideNavItems = views[userRole]

      roleSideNavItems.forEach(({ id: roleSideNavItemId, children }, index) => {
        const sideNavBarItemIndex = availableSideNavItems.findIndex(
          ({ id }) => id === roleSideNavItemId
        )

        // ... if the item is already on the availableSideNavItems array, just merge the children
        if (sideNavBarItemIndex > -1) {
          availableSideNavItems[sideNavBarItemIndex].children?.push(...(children ?? []))
        }

        // ... if not add the side nav item inside the availableSideNavItems array result.
        if (sideNavBarItemIndex === -1) {
          availableSideNavItems.push(roleSideNavItems[index])
        }
      })
    })

    availableSideNavItems.push({ routeKey: 'NOTIFICATION', StartIcon: EmailIcon })

    if (currentRoles.includes('admin')) {
      availableSideNavItems.push({ routeKey: 'PARTY_REGISTRY' as RouteKey })
    }

    // Remove duplicated children, if there's any.
    availableSideNavItems.forEach((_, index) => {
      if (availableSideNavItems[index]?.children) {
        availableSideNavItems[index].children = uniq(availableSideNavItems[index].children)
      }
    })

    return availableSideNavItems
  }, [currentRoles])
}
