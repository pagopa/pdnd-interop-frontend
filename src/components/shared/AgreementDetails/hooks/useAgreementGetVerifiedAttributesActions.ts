import { isAttributeOwned, isAttributeRevoked } from '@/utils/attribute.utils'
import { useAgreementDetailsContext } from '../AgreementDetailsContext'
import { useTranslation } from 'react-i18next'
import { useCurrentRoute } from '@/router'
import { useJwt } from '@/hooks/useJwt'
import { AttributeMutations } from '@/api/attribute'
import type { AttributeContainer } from '@/components/layout/containers'

/**
 * Returns the actions for the verified attributes section inside the agreement details.
 * @returns A function that returns an array of actions for the given attribute.
 */
export const useAgreementGetVerifiedAttributesActions = () => {
  const { t } = useTranslation('agreement', { keyPrefix: 'read.attributes' })
  const { mode } = useCurrentRoute()
  const { isAdmin } = useJwt()
  const { partyAttributes, isAgreementEServiceMine, agreement } = useAgreementDetailsContext()

  const { mutate: verifyAttribute } = AttributeMutations.useVerifyPartyAttribute()
  const { mutate: revokeAttibute } = AttributeMutations.useRevokeVerifiedPartyAttribute()

  const ownedVerifiedAttributes = partyAttributes?.verified ?? []

  return (attributeId: string) => {
    // The user can certify verified attributes in this view only if it is a provider...
    if (!agreement || mode === 'consumer' || !isAdmin) return []
    // ... only if the e-service does not belong to itself
    if (isAgreementEServiceMine) return []
    // ... and only if the agreement is active, pending or suspended
    if (!['ACTIVE', 'PENDING', 'SUSPENDED'].includes(agreement.state)) return []

    const attribute = ownedVerifiedAttributes.find((a) => a.id === attributeId)

    const isOwned = isAttributeOwned('verified', attributeId, ownedVerifiedAttributes)
    const isOwnedButRevoked = attribute && isAttributeRevoked('verified', attribute)

    const handleVerifyAttribute = (attributeId: string) => {
      verifyAttribute({
        partyId: agreement.consumer.id,
        id: attributeId,
      })
    }

    const handleRevokeAttribute = (attributeId: string) => {
      revokeAttibute({
        partyId: agreement.consumer.id,
        attributeId,
      })
    }

    const attributeActions: React.ComponentProps<typeof AttributeContainer>['actions'] = [
      {
        label: isOwned ? t('verified.actions.update') : t('verified.actions.verify'),
        action: handleVerifyAttribute,
      },
    ]

    if (isOwned && !isOwnedButRevoked) {
      attributeActions.push({
        label: t('verified.actions.revoke'),
        action: handleRevokeAttribute,
        color: 'error',
      })
    }

    return attributeActions
  }
}
