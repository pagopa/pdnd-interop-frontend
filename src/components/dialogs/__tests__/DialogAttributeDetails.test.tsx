import React from 'react'
import { DialogAttributeDetails } from '../DialogAttributeDetails'
import { setupServer } from 'msw/node'
import { rest } from 'msw'
import { BACKEND_FOR_FRONTEND_URL } from '@/config/env'
import type { Attribute } from '@/api/api.generatedTypes'
import { renderWithApplicationContext } from '@/utils/testing.utils'
import { waitFor } from '@testing-library/react'
import { vi } from 'vitest'
import * as useJwtHook from '@/hooks/useJwt'
import { mockUseJwt } from '__mocks__/data/user.mocks'

vi.spyOn(useJwtHook, 'useJwt').mockImplementation(() => mockUseJwt())

const server = setupServer(
  rest.get(`${BACKEND_FOR_FRONTEND_URL}/attributes/:id`, (req, res, ctx) => {
    return res(
      ctx.json({
        id: 'id',
        name: 'name',
        kind: 'CERTIFIED',
        description: 'description',
        creationTime: '2021-01-01T00:00:00Z',
      } satisfies Attribute)
    )
  })
)

beforeAll(() => server.listen())
afterAll(() => server.close())

describe('DialogAttributeDetails testing', () => {
  it('should match the snapshot', async () => {
    const screen = renderWithApplicationContext(
      <DialogAttributeDetails
        type="showAttributeDetails"
        attribute={{
          id: 'id',
          name: 'name',
        }}
      />,
      { withReactQueryContext: true }
    )
    expect(screen.baseElement).toMatchSnapshot('loading state')
    await waitFor(() => expect(screen.getByText('description')).toBeDefined())
    expect(screen.baseElement).toMatchSnapshot('full state')
  })
})
