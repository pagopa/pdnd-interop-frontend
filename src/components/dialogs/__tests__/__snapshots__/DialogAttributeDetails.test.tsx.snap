// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DialogAttributeDetails testing > should match the snapshot > full state 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r0:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id=":r0:"
        >
          name
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <div
            class="css-1gw2iun-MuiStack-root"
          >
            <div
              class="css-nen11g-MuiStack-root"
            >
              <div
                class="MuiBox-root css-1pdb85g"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  content.descriptionField.label
                </p>
              </div>
              <div
                class="MuiBox-root css-1rr4qq7"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                >
                  description
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            closeBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`DialogAttributeDetails testing > should match the snapshot > loading state 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r0:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id=":r0:"
        >
          name
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <div
            class="css-1gw2iun-MuiStack-root"
          >
            <div
              class="css-19qjufr-MuiStack-root"
            >
              <div
                class="MuiBox-root css-iuqs4u"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
              </div>
              <div
                class="MuiBox-root css-1rr4qq7"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            closeBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
